### ISSUE TITLE
data overlap

### ISSUE THREAD

**Issue by Daybreak-Zheng** [SadilKhan/Text2CAD#6]
Great job. We found that there is an overlap between the training and testing data, especially at the abstract level.

1. One abstract text corresponds to multiple cad model sequences. Does this affect the model training?
2. Will including the test data in the training data lead to data leakage and inaccurate testing metrics?

Looking forward to your reply.

**Comment by SadilKhan** [comment]
Hello @Daybreak-Zheng,

1. Multiple CAD sequences from a single text: This is a desirable quality of CAD dataset. This helps modeling design intent as designers often have unique styles of designing. This is a one-to-many problem (similar to text-to-image generation). For this type of problem in CAD setting, it's best to use Autoregressive models rather than Non-Autoregressive which can handle such tasks. This setup does not negatively affect the model's training.

2. Abstract-level descriptions are inherently simple and often lack detailed specificity. For instance, a description like "two rectangular prisms" could correspond to either `two entirely different CAD models (bigger vs smaller) with distinct CAD sequences` or `identical CAD models generated through different sequences`. For this project, we used the original train and test split, proposed in DeepCAD. However, I welcome anyone who can generate a clean train and test split. This would also require retraining the DeepCAD autoencoder from scratch for comparison purposes.

**Comment by Daybreak-Zheng** [comment]
Thank you very much for your reply. 
I would like to ask about how the model is trained，It seems that I did not find these contents in the paper.

For multiple different text levels, is the text of the three levels concatenated together? 
Or is only one text level used for training at a time? What is the training proportion for different levels? 

I am looking forward to your reply. 
<img width="757" alt="Clipboard_Screenshot_1735132046" src="https://github.com/user-attachments/assets/d4f11276-32a0-4c13-91ab-d7bbc1ab70f7" />

**Comment by SadilKhan** [comment]
Hello,

- During training, only one text is used at a time. The four texts in the figure are only for visualization purposes. 
- In the original paper, we created four new training pairs for each CAD model which gave us $\sim 600$k training samples. So, all the levels were used in each epoch.

However, I am doing another experiment where I am taking one level randomly in each epoch and then I will check if it has any noticeable performance drop/gain. The dataloader code I shared in this repo contains the latter setting but can be modified for the earlier one as well.

**Comment by Daybreak-Zheng** [comment]
Thank you very much for your reply.
I still have some questions. 
Did you set **the max seq len** for text descriptions at different levels during training? 
Since the instructions at **the expert level are very long**, did you set the maximum input length to 2048 or 4096?
I am looking forward to your reply.

**Comment by SadilKhan** [comment]
Hello,
The maximum input length for texts is 512.

**Comment by Daybreak-Zheng** [comment]
Thank you very much for your reply.
Will setting the maximum length to 512 cause many text descriptions to be truncated and thus affect the performance? Because I've noticed that there are a great many text descriptions at the expert level.

**Comment by SadilKhan** [comment]
Yes This will actually truncate it however I do not think it will impact a lot. We use 512 since during the data annotation, we set the limit of the expert levels for LLM to 512. So the average word limit of expert levels should be around 512 because LLMs do not always adhere exactly to the word limit. 

I think a better option would be to be dynamic about the word length and take the maximum number of words per batch. You can set it [here](https://github.com/SadilKhan/Text2CAD/blob/main/Cad_VLM/models/layers/text_embed.py#L39).
